default.nix

```diff
@@ -30,21 +30,27 @@ rec {
           path_ = toString f;
           path__ = "${rootStr}/${f}";
         in
-        if builtins.isFunction f then f
-        else if builtins.isPath f then (path: _: path_ == path)
-        else if builtins.isString f then (path: _: path__ == path)
+        if builtins.isFunction f then
+          builtins.concatMap (p: _expandPath root p []) (_matchManually root f)
+        else if builtins.isPath f then _expandPath root f []
```

Comment:
Once the other PR is merged, I think this could be rewritten in terms of `path_ == path || hasPrefix "${path_}/" path` and remove all of the rest.